storage/src/main/java/org/apache/kafka/storage/internals/log/LogValidator.java

```diff
@@ -275,7 +278,7 @@ public ValidationResult assignOffsetsNonCompressed(LongRef offsetCounter,
 
             if (batch.magic() > RecordBatch.MAGIC_VALUE_V0 && maxBatchTimestamp > maxTimestamp) {
                 maxTimestamp = maxBatchTimestamp;
-                offsetOfMaxTimestamp = offsetOfMaxBatchTimestamp;
+                shallowOffsetOfMaxTimestamp = offsetOfMaxBatchTimestamp;
```

Comment:
We need to set shallowOffsetOfMaxTimestamp to the last offset in the batch. Do we have test cases covering that?

Reply:
> Do we have test cases covering that?

There is a UT already, and I will add more tests to cover it.

https://github.com/apache/kafka/blob/cc6b919212ae62d75850214ae2c93379b78ff325/core/src/test/scala/unit/kafka/log/LogValidatorTest.scala#L408